src/DynamoCoreWpf/Views/Preview/Watch3DView.xaml.cs

```diff
@@ -215,5 +217,26 @@ private void view_MouseRightButtonUp(object sender, MouseButtonEventArgs e)
         }
 
         #endregion
+
+        #region interface methods
+
+        public Ray3D GetClickRay(MouseEventArgs mouseButtonEventArgs)
+        {
+            var mousePos = mouseButtonEventArgs.GetPosition(this);
+
+            return View.Point2DToRay3D(new Point(mousePos.X, mousePos.Y));
+        }
+
+        public void AddGeometryFromRenderPackages(IEnumerable<IRenderPackage> packages)
```

Comment:
Because these methods just call methods on the view model, I propose instead of implementing the `IWatch3DView` interface on the view, you do it on the view model. So, make it `IWatch3DViewModel`. Make `Watch3DViewModelBase : IWatch3DViewModel`, and get access to the one you want through the `DynamoViewModel.Watch3DViewModels` property.